subsys/spm/Kconfig

```diff
@@ -66,6 +76,21 @@ config SPM_SECURE_SERVICES
 	  Note: Please set this and SPM_SERVICE_* configs from the app, instead
 	  of the SPM. This ensures that the values are in sync between the two
 	  images.
+
+config SPM_NS_DEBUG_LOGS_DUMP
+	bool "Enable dumping non-secure logs on secure fault"
+	help
+	  When enabled secure fault handler is calling non-secure function to
+	  synchronously dump all pending logs. It must be enabled in spm and
+	  in non-secure application. Feature is only for debugging purposes and
+	  is considered a security violation. SPM_SHARE_CONSOLE_UART must be
+	  enabled in spm.
```

Comment:
> SPM_SHARE_CONSOLE_UART must be enabled in spm.

From what I can tell, SPM_SHARE_CONSOLE_UART isn't really needed. To me it seems like SPM_SHARE_CONSOLE_UART and SPM_NS_DEBUG_LOGS_DUMP are completely independent.

Reply:
when UARTE is set to non-secure and locked before jumping to ns then when secure fault handler occurs it jumps to spm and attempts to access non-secure UARTE providing secure buffer pointer. It seems that it generates another secure fault since when `SPM_SHARE_CONSOLE_UART` is disabled i see only reset and no data printed.

Reply:
SPM_NS_DEBUG_LOGS_DUMP is only for dumping NS logs right?